rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,220 @@
+- Also, users should have migrated to 0.16.x before upgrading to 1.x. We do not want to add read support for very old versions of hudi in 1.x(for eg 0.7.0). 
+- So, in an effort to bring everyone to latest hudi versions, 1.x reader will have full read capabilities for 0.16.x, but for older hudi versions, 1.x reader may not have full reader support. 
+The reocmmended guideline is to upgrade all readers and writers to 0.16.x. and then slowly start upgrading to 1.x(readers followed by writers). 
+
+Before we dive in further, lets understand the format changes:
+
+## Format changes
+### Table properties
+- Payload class ➝ payload type.
+- New metadata partitions could be added (optionally enabled)
+
+### MDT changes
+- New MDT partitions are available in 1.x. MDT schema upgraded.
+- RLI schema is upgraded to hold row position
+
+### Timeline:
+- [storage changes] Completed write commits have completed times in the file name.
+- [storage changes] Completed and inflight write commits are in avro format which were json in 0.x.
+- We are switching the action type for clustering from “replace commit” to “cluster”.
+- Similarly, for completed compaction, we are switching from “commit” to “compaction” in an effort to standardize actions for a given write operation.
+- [storage changes] Timeline ➝ LST timeline. There is no archived timeline in 1.x
+- [In-memory changes] HoodieInstant changes due to presence of completion time for completed HoodieInstants.
+
+### Filegroup/FileSlice changes:
+- Log files contain delta commit time instead of base instant time.
+- Log appends are disabled in 1.x. In other words, each log block is already appended to a new log file.
+- File Slice determination logic for log files changed (in 0.x, we have base instant time in log files and its straight forward. In 1.x, we find completion time for a log file and find the base instant time (parsed from base files) which has the highest value lesser than the completion time of the log file).
+- Log file ordering within a file slice. (in 0.x, we use base instant time ➝l log file versions ➝ write token) to order diff log files. in 1.x, we will be using completion time to order).
+
+### Log format changes:
+- We have added new header types in 1.x. (IS_PARTIAL)
+
+## Changes to be ported over 0.16.x to support reading 1.x tables
+### What will be supported
+- For features introduced in 0.x, and tables written in 1.x, 0.16.0 reader should be able to provide consistent reads w/o any breakage.
+### What will not be supported
+- A 0.16 writer cannot write to a table that has been upgraded-to/created using 1.x without downgrading to 0.16
+- For new features introduced in 1.x, we might call out that 0.16.x reader may not support reads.
+- if deletion vector is enabled, 0.16.0 reader should fallback to key based merges, giving up on the performance optimization. 
+- if partial updates are enabled, 0.16.0 reader should fallback to key based merges, giving up on the performance optimization.
```

Comment:
Partial updates need new merging logic so it may not be supported without porting the (key-based) partial merging logic to 0.16.

Reply:
sg. the fallback support needs to be ported to 0.16. will update the RFC. is it the same for deletion vector as well ?